src/routing/comm.rs

```diff
@@ -136,30 +141,7 @@ impl Comm {
 
     /// Tests whether the peer is reachable.
     pub async fn is_reachable(&self, peer: &SocketAddr) -> Result<(), Error> {
-        let qp2p_config = qp2p::Config {
-            local_ip: Some(self.endpoint.local_addr().ip()),
-            local_port: Some(0),
-            forward_port: false,
-            ..Default::default()
-        };
-
-        let qp2p = QuicP2p::with_config(Some(qp2p_config), &[], false)
-            .map_err(|err| Error::InvalidConfig { err })?;
-        let (connectivity_endpoint, _, _, _) = qp2p
-            .new_endpoint()
-            .await
-            .map_err(|err| Error::CannotConnectEndpoint { err })?;
-
-        connectivity_endpoint
-            .is_reachable(peer)
-            .await
-            .map_err(|err| {
-                info!("Peer {} is NOT externally reachable: {}", peer, err);
-                Error::AddressNotReachable { err }
-            })
-            .map(|()| {
-                info!("Peer {} is externally reachable.", peer);
-            })
+        is_reachable(self.endpoint.local_addr().ip(), peer).await
```

Comment:
Why do we move this logic to a helper function? the body is not that large and we are not breaking up anyway. Also, I'm getting a weird error due to this as it seems something becomes not `Send`